### PR TITLE
ci: introduce ci/docker/env.sh

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

context: https://discord.com/channels/428295358100013066/560503042458517505/1310571536221995068

it's time to update our ci to ubuntu 22

#### Summary of Changes

- move all docker related env into `ci/docker/env.sh`
- use a new tag format for our ci docker image.
```
CI_DOCKER_IMAGE="anzaxyz/ci:${SANITIZED_BASE_IMAGE}_rust-${RUST_VERSION}_${RUST_NIGHTLY_VERSION}_${ci_docker_hash}"`
```

### CODE REVIEW

**Comment by brooksprumo** [comment]
I'm not sure how to review this for correctness nor security, so I'm removing myself as a reviewer.

**Comment by anza-team** [comment]
:scream: New commits were pushed while the automerge label was present.

**Review by sakridge — APPROVED** [review]

**Review by willhickey — APPROVED** [review]

File: net/net.sh [hunk 1]
```diff
@@ -192,9 +192,9 @@ build() {
   supported=("20.04")
```
Comment (t-nelson) [inline net/net.sh:192]: does this need to change?

Comment (yihau) [reply net/net.sh:192]: good catch! 🕵️ https://github.com/anza-xyz/agave/pull/3792

**Pull request merged**